lib/web/websocket/receiver.js

```diff
@@ -298,26 +298,28 @@ class ByteParser extends Writable {
       // defined as the status code (Section 7.4) contained in the first Close
       // control frame received by the application
       code = data.readUInt16BE(0)
+      if (!isValidStatusCode(code)) {
+        return null
+      }
+    } else {
+      return null
```

Comment:
A close frame doesn't need to contain a reason or code, `data` (the body) can be empty here. You could assert the data length isn't 1 here, but we do check for its length elsewhere so it'd just be for safety.

Reply:
The thing is: 
https://app.codecov.io/gh/nodejs/undici/pull/3080/blob/lib/web/websocket/receiver.js

This line is covered by test. So it is possible that the size of the data is 0.

Reply:
I know, I said that. The buffer here is not the entire frame (which has to have a minimum length of 2), but the body of a frame. If there is no code and no reason, it is still a valid close frame, and will be empty here. Returning `null` means that the frame body is invalid, but that's not correct.

Reply:
On the other hand, the body can be empty or 2+ bytes, but it cannot be 1 byte, which is why I mentioned earlier that for safety we could assert that the body isn't a single byte. We already check elsewhere, but as I mentioned earlier, would be for safety.

Reply:
Well but it is happening:

![image](https://github.com/nodejs/undici/assets/5059100/c100e62c-52d0-4da0-a7b0-668efc96aeae)

Reply:
I don't understand what that has to do with what I said?

Reply:
This means, that your claim, that we already handled all cases with data.length < 2 is invalid. Either data.length === 0 or data.length === 1 or both are not covered by the calling function.